cmake/usage/kconfig-usage.cmake

```diff
@@ -1,15 +1,2 @@
-message(" config          - Update current config utilising a line-oriented program")
-message(" nconfig         - Update current config utilising a ncurses menu based program")
-message(" menuconfig      - Update current config utilising a menu based program")
-message(" xconfig         - Update current config utilising a QT based front-end")
-message(" gconfig         - Update current config utilising a GTK based front-end")
-message(" oldconfig       - Update current config utilising a provided .config as base")
-message(" silentoldconfig - Same as oldconfig, but quietly, additionally update deps")
-message(" defconfig       - New config with default from ARCH supplied defconfig")
-message(" savedefconfig   - Save current config as ./defconfig (minimal config)")
-message(" allnoconfig     - New config where all options are answered with no")
-message(" allyesconfig    - New config where all options are accepted with yes")
-message(" alldefconfig    - New config with all symbols set to default")
-message(" randconfig      - New config with random answer to all options")
-message(" listnewconfig   - List new options")
-message(" olddefconfig    - Same as silentoldconfig but sets new symbols to their default value")
+message(" menuconfig      - Update current .config or generate a minimal configuration")
+message(" oldconfig       - Update an old .config file, prompting for the values of new options")
```

Comment:
what's the point of "oldconfig" anymore? this uses `conf`, which we want to get rid of?

Reply:
It's a common tool on Linux when you have a working `.config` that you want to update to a new kernel version at least. There's a Zephyr issue for it as well: https://github.com/zephyrproject-rtos/zephyr/issues/5426 (that's what prompted me to add a Kconfiglib version, originally :).

I don't know how much different tools are used in practice in Zephyr though.